screens/WelcomeScreenController.ts

```diff
@@ -55,9 +63,18 @@ export function useWelcomeScreen(props: RootRouteProps) {
     unlockPage: () => {
       // prioritize biometrics
       if (!isSettingUp && isBiometricUnlockEnabled && biometrics !== '') {
-        props.navigation.navigate('Biometric', { setup: isSettingUp });
+        props.navigation.navigate('Biometric', {setup: isSettingUp});
       } else if (!isSettingUp && passcode !== '') {
-        props.navigation.navigate('Passcode', { setup: isSettingUp });
+        sendStartEvent(getStartEventData('App Login'));
+        sendInteractEvent(
+          getInteractEventData(
+            'App Login',
+            'TOUCH',
+            'Unlock application button',
+          ),
+        );
+        sendImpressionEvent(getImpressionEventData('App Login', 'Passcode'));
+        props.navigation.navigate('Passcode', {setup: isSettingUp});
```

Comment:
Impression Event should be tracked once we are on passcode screen

Reply:
done